File: 프로그래머스/lv2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
  const rest = []
  const result = []
  
  for (let i = 0; i < progresses.length; i++) {
    const restEl = Math.ceil((100 - progresses[i]) / speeds[i])
    rest.push(restEl)
  }
  
  let count = 1
  let maxDay = rest[0]
  
  for (let j = 1; j < rest.length; j++) {
    if (maxDay >= rest[j]) {
      count += 1
    } else {
      result.push(count)
      maxDay = rest[j]
      count = 1
    }
  }
  
  result.push(count)
  
  return result
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math

def solution(progresses, speeds):
    # 진도가 100일 때 서비스 반영
    # 앞에 있는 기능이 배포될 때 함께 배포
    # 큐...?
    # 각 기능이 며칠 작업 후 배포가 가능한지 큐에 저장
    queue = []
    
    for p, s in zip(progresses, speeds):
        days = math.ceil((100 - p) / s) # 추가 작업 시간
        queue.append(days)
    
    prev = queue[0]
    count = 1
    result = []
    
    for q in queue[1:]:
        if prev >= q:
            count += 1
        else:
            result.append(count)
            count = 1
            prev = q
    result.append(count)
    return result
    
    
        